### PR TITLE
Add threads to finalizers that have list of objects.

### PR DESCRIPTION
**Pull request by shyRozen**
Today the finalizer are acting in serial if there are instances of the same object kind. The teardown takes as long as the test, and even more. This PR is intent to make a finalizer that has a list of objects of the same kind to delete them in parallel.
Signed-off-by: Shay Rozen <shay.rozen@gmail.com>

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -5682,3 +5719,20 @@ def teardown():
     workload.deploy_workload()
 
     return workload
+
+
+def thread_finalizer(operation, instances):
```
Comment (ebenahar) [inline tests/conftest.py:5724]: not all places that call this function use it as a finalizer. 
I can see that for instance, [here](https://github.com/red-hat-storage/ocs-ci/pull/6159/files#diff-e52e4ddd58b7ef887ab03c04116e676f6280b824ab7469d5d3080e5cba4f2128R845), you are calling it from a non teardown fixture function. 
Maybe woth representing that fact by changing the function name from  `thread_finalizer` to `thread_operation`

Comment (shyRozen) [reply tests/conftest.py:5724]: Sounds good

Comment (shyRozen) [reply tests/conftest.py]: Changed

**Comment by shyRozen** [comment]
To ambitious. Lowering the resource in #6250

**Pull request closed without merging**